### PR TITLE
Filter out duplicates during inventory collection

### PR DESCRIPTION
**Pull request by mansam**
Initial fix for https://bugzilla.redhat.com/show_bug.cgi?id=1528663

There are some disadvantages that make it difficult to fix the bug by changing pagination in openstack_handle::pagination::marker.rb, since pagination can be different for each service and collection, Fog can make it difficult to get the raw response, and some Fog collections behave differently from others. To avoid these problems, this fix implements a filter for uniqueness at the point where the inventory is collected.

Long term the correct fix is to totally reimplement the way that paging is done in OpenstackHandle/Fog.

### CODE REVIEW

File: app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@ def availability_zones
 
   def cloud_services
     return @cloud_services if @cloud_services.any?
-    @cloud_services = compute_service.handled_list(:services, {}, openstack_admin?)
+    @cloud_services = uniques(compute_service.handled_list(:services, {}, openstack_admin?))
```
Comment (Ladas) [inline app/models/manageiq/providers/openstack/inventory/collector/cloud_manager.rb]: we can ignore graph refresh, its indexes will not allow duplicate records to be passed to persistor

**Comment by Ladas** [comment]
@mansam we will need to rerecord the VCRs, conflict with the storage targeted refresh?

**Comment by mansam** [comment]
@Ladas Probably. They're currently rerecording.

**Comment by aufi** [comment]
Looks good to me, merging!

**Comment by mansam** [comment]
@Ladas Yeah, specs to reproduce this are on my log of things to do.

**Comment by mansam** [comment]
@miq-bot add_label fine/yes

**Comment by simaishi** [comment]
@mansam For backporting to Gaprindashvili branch, all VCR files are conflicting on cherry-pick. If I should take all files from this PR ignoring the conflict, I can handle that. Otherwise, please create a PR for Gaprindashvili.

For backporting to Fine branch, there are only 3 files that are clean cherry-pick. Please create a PR for Fine branch.

**Comment by simaishi** [comment]
Backported to Gaprindashvili via https://github.com/ManageIQ/manageiq-providers-openstack/pull/219

**Comment by simaishi** [comment]
Backported to Fine via https://github.com/ManageIQ/manageiq/pull/16934

**Pull request merged**